microservices/first/utils/db_manager.py

```diff
@@ -0,0 +1,127 @@
+import json
+import sqlite3
+from sqlite3 import Error
+from typing import Any, List, Optional
+
+class DBManager:
+  def __init__(self, db_file: str) -> None:
+    """
+    Initialize the database manager with the path to the database file.
+    
+    param db_file: The file path of the SQLite database.
+    """
+    self.db_file = db_file
+    self.conn = self.create_connection()
+
+  def create_connection(self) -> Optional[sqlite3.Connection]:
+    """Create a database connection to the SQLite database."""
+    conn = None
+    try:
+        conn = sqlite3.connect(self.db_file)
+        return conn
+    except Error as e:
+        print(e)
+    return conn
+
+  def close_connection(self) -> None:
+    """Close the database connection."""
+    if self.conn:
+        self.conn.close()
+
+  def create_deployments_table(self) -> None:
+    """Create the deployments table in the database."""
+    try:
+        c = self.conn.cursor()
+        c.execute(
+                  '''CREATE TABLE IF NOT EXISTS deployments (
+                  id INTEGER PRIMARY KEY, 
+                  repo_name TEXT, 
+                  repo_branch TEXT,
+                  status TEXT DEFAULT 'STARTED')'''
+                )
+    except Error as e:
+        print(e)
+
+  def create_results_table(self) -> None:
+    """Create the results table in the database."""
+    try:
+        c = self.conn.cursor()
+        c.execute(
+            '''CREATE TABLE IF NOT EXISTS results (
+                id INTEGER PRIMARY KEY,
+                deployment_id INTEGER,
+                result TEXT,
+                FOREIGN KEY(deployment_id) REFERENCES deployments(id)
+            )'''
+        )
+    except Error as e:
+        print(e)
+
+  def add_result(self, deployment_id: int, result: str) -> int:
+    """Add a new result to a specific deployment."""
+    sql = '''INSERT INTO results(deployment_id, result) VALUES(?, ?)'''
+    cur = self.conn.cursor()
+    cur.execute(sql, (deployment_id, result))
+    self.conn.commit()
+    return cur.lastrowid
+  
+  def add_results(self, deployment_id: int, results: list) -> None:
+    """Add multiple results to a specific deployment."""
+    sql = '''INSERT INTO results(deployment_id, result) VALUES(?, ?)'''
+    cur = self.conn.cursor()
+    for result in results:
+        cur.execute(sql, (deployment_id, result))
+    self.conn.commit()
+
+  def get_results(self, deployment_id: int) -> list:
```

Comment:
Can specify types in lists

```suggestion
  def get_results(self, deployment_id: int) -> list[str]:
```